src/app/clusters/window-covering-server/window-covering-server.h

```diff
@@ -139,6 +139,14 @@ void TiltPositionSet(chip::EndpointId endpoint, NPercent100ths position);
 
 EmberAfStatus GetMotionLockStatus(chip::EndpointId endpoint);
 
+/**
+ * @brief PostAttributeChange is called when an Attribute is modified
+ *
+ * @param[in] endpoint
+ * @param[in] attributeId
+ */
+void PostAttributeChange(chip::EndpointId endpoint, chip::AttributeId attributeId);
```

Comment:
I guess, but why is the _app_ implementing the _cluster_ attribute-changed callback, not the _app_ attribute-changed callback, necessitating this call from the app back into the cluster to notify it about the fact that the attribute change it should have known about anyway has happened?

At the very least, this interaction, which is not how other clusters work, needs to be very clearly documented here.  Because nothing in the docs there says "the app must call this function"....

Reply:
Yeah, I agree and honestly I was also concerned about that, when analyzing the bug root cause. However it was more about fixing the certification fault from my side and I didn't want to modify the cluster code that I don't fully understand much. @jmeg-sfy could we consider changing the cluster behavior in some followup PR?

Reply:
Yeah this should be changed in the future. This post Attribute change function is originally there to detect internal/external writes on target position attributes.
it is weak because the implementation of all clusters is different than the original one on EFR32.
We should try to converge to a common a codebase abstracted from the platform rather than forking.
I would support any PR going into this way. I don't know the implementation of NRF well need to look at it.